### ISSUE TITLE
No message shown on file upload error

### ISSUE THREAD

**Issue by svivian** [bolt/bolt#5589]
If a file cannot be uploaded, in some places no error is shown. To reproduce:
1. Set `public/files` directory to not be writable by the server.
2. Go edit a page and click "Upload" in the stack to the right.
3. Try to upload a file.

The file starts uploading with a progress bar, but when it's done nothing happens - the file is not actually saved anywhere and it doesn't appear in the stack. There is no error message.

Next:
1. Go to the File Management page.
2. Try to upload a file.

I got an RuntimeException (Fatal Error) - "Unable to write to upload destination. Check permissions on .../public/files". Should this be a fatal error? I would expect to be sent back to the file manager with the above error message (in the usual red box... I think you call it "flashbag"?), and not for the app to crash.

Next:
1. Set the `public/files` folder to be writable (files now upload to that specific directory fine).
2. Add a subfolder _without_ the correct permissions (I added it in Finder as I forgot there was a button for it in Bolt).
3. Try to upload a file.

Now I get no error message at all, and no file is uploaded.

So in summary:
- No error message when a file cannot be uploaded via the stack.
- Fatal Error when a file cannot be uploaded in the root of the File Manager, instead of a "flashbag".
- No error message when a file cannot be uploaded to a subfolder in the File Manager, when the root folder has correct permissions but the subfolder does not.
### Details
- Bolt 3.0.11
- PHP 5.5
- Apache server
- Chrome browser, Mac

**Comment by Maelstromeous** [comment]
This really needs addressing. It's so confusing for users when they see an upload indicator but then absolutely nothing when the upload fails. :'(

**Comment by SahAssar** [comment]
Closing as a dupe of #4895

**Comment by bobdenotter** [comment]
A potential cause for this is if the current URL doesn't match the canonical. A warning to help the user identify this, is set here: https://github.com/bolt/configuration-notices/blob/master/src/EventListener/ConfigurationNoticesListener.php#L247
